### PR TITLE
Remove defaults for topic and partition

### PR DESCRIPTION
**Pull request by dantswain**
This was masking at least one bug - the partition was not being passed through to the offset commit request in Server.fetch, which would probably cause offset commits to always be commit to partition 0.

This looks like a lot of changes, but most of the changes are either removing the default values from structs or updating the tests where they were previously relying on the defaults.  I also tried to update documentation wherever it was missing values for partition or topic.

I would like to do a similar PR for the default "kafka_ex" consumer group, which I believe is masking similar bugs as per #69.  I'll wait until #71 is resolved for that.

### CODE REVIEW

File: lib/kafka_ex/protocol/offset_commit.ex [hunk 1]
```diff
@@ -1,11 +1,11 @@
 defmodule KafkaEx.Protocol.OffsetCommit do
   defmodule Request do
-    defstruct consumer_group: "kafka_ex", topic: "", partition: 0, offset: 0, metadata: ""
+    defstruct consumer_group: "kafka_ex", topic: nil, partition: nil, offset: nil, metadata: ""
```
Comment (bjhaid) [inline lib/kafka_ex/protocol/offset_commit.ex]: `kafka_ex` should go away too

File: lib/kafka_ex/protocol/offset_fetch.ex [hunk 2]
```diff
@@ -1,11 +1,11 @@
 defmodule KafkaEx.Protocol.OffsetFetch do
   defmodule Request do
-    defstruct consumer_group: "kafka_ex", topic: "", partition: 0
+    defstruct consumer_group: "kafka_ex", topic: nil, partition: nil
```
Comment (bjhaid) [inline lib/kafka_ex/protocol/offset_fetch.ex]: same here too

File: lib/kafka_ex/protocol/offset_commit.ex (hunk 1 above)
Comment (dantswain) [inline lib/kafka_ex/protocol/offset_commit.ex]: I agree.  I wanted to do that in a separate PR because it has some other implications.
